tests/samplers_tests/test_nsgaii.py

```diff
@@ -115,6 +115,22 @@ def test_swapping_prob() -> None:
         NSGAIISampler(swapping_prob=1.1)
 
 
+@pytest.mark.parametrize("categorical_values1", [[1, 2, 3], [True, False], ["True", "False"]])
+@pytest.mark.parametrize("categorical_values2", [[10, 12], ["-2", "-1"]])
```

Comment:
Thank you for your indication.
I think using two different types of categorical_values was essential for the error (https://github.com/optuna/optuna/issues/3982#issuecomment-1263194047), but I agree that it will be more simple to write test cases without using combination. I fixed it accordingly.

Reply:
Hmm, the new version of the test function has more lines than before.

> I think using two different types of categorical_values was essential for the error (https://github.com/optuna/optuna/issues/3982#issuecomment-1263194047)

The minimum requirement of the test case seems to be a combination of raw values and stringified values.
If so, how about just parametrizing raw values? I guess stringified values can be generated in the test case.
The following is just an example.

```python
@pytest.mark.parametrize("choices", [[-1, 0, 1], [True, False], ["string1", "string2"]])
def test_crossover_casting(choices: List[Any]) -> None:
    def objective(trial: optuna.Trial) -> Sequence[float]:
        str_choices = list(map(str, choices))
        r = trial.suggest_categorical("r", choices)
        s = trial.suggest_categorical("s", str_choices)
        assert isinstance(r, type(choices[0]))
        assert isinstance(s, type(str_choices[0]))
        return 1.0, 2.0
```

This test case lacks the pair of numbers and bools, but I don't think it is critical because the bug just requires two different types of values (e.g., numbers and strings, or bools and strings).

Reply:
Thank you for applying the suggestion to the PR.
@HideakiImamura What do you think of this simplification?

Reply:
Looks very good.